cli/src/commands/start.rs

```
@@ -313,20 +313,24 @@ impl Start {
     fn parse_trusted_peers(&self) -> Result<Vec<SocketAddr>> {
         let Some(peers) = &self.peers else { return Ok(vec![]) };
 
-        // Split on an empty string returns an empty string.
-        if peers.is_empty() {
-            return Ok(vec![]);
-        }
-
-        let mut result = vec![];
-        for ip in peers.split(',') {
-            match ip.parse::<SocketAddr>() {
-                Ok(ip) => result.push(ip),
-                Err(err) => bail!("An address supplied to --peers ('{ip}') is malformed: {err}"),
-            }
+        match peers.is_empty() {
+            // Split on an empty string returns an empty string.
+            true => Ok(vec![]),
+            false => Ok(peers
+                .split(',')
+                .flat_map(|ip_or_hostname| {
+                    let trimmed = ip_or_hostname.trim();
+                    match trimmed.to_socket_addrs() {
+                        Ok(mut ip_iter) => {
+                            // A hostname might resolve to multiple IP addresses. We will use only the first one,
+                            // assuming this aligns with the user's expectations.
+                            ip_iter.next()
```

If the socket address does not resolve, does this just fail silently? Might be good to print a warning if `ip_iter.next()` returns `None`.

---

Good point, I'd argue it's not a nit :) 003bae5b4